quickwit/quickwit-indexing/src/split_store/local_split_store.rs

```diff
@@ -138,22 +134,17 @@ impl LocalSplitStore {
             }
         }
 
-        if split_files.len() > params.max_num_splits {
-            return Err(StorageErrorKind::InternalError.with_error(anyhow::anyhow!(
-                "Initial number of files exceeds the maximum number of files allowed.",
-            )));
-        }
-
-        if total_size_in_bytes > params.max_num_bytes {
-            return Err(StorageErrorKind::InternalError.with_error(anyhow::anyhow!(
-                "Initial cache size exceeds the maximum size in bytes allowed.",
-            )));
-        }
-
+        let mut cache_size_checker_lock = cache_size_checker.lock().await;
+        cache_size_checker_lock.add_initial_splits(
+            &local_storage_root,
+            split_files.len(),
+            total_size_in_bytes,
+        )?;
+        drop(cache_size_checker_lock);
```

Comment:
Why call `drop` explicitly?

Reply:
The `LocalSplitStore` struct needs to take ownership of `cache_size_checker` which is the parent of the lock.